### PR TITLE
Extend "Create External CLuster Endpoint" for KubeOne Clusters

### LINKED ISSUE #9284: Extend "Create External CLuster Endpoint" for KubeOne Clusters
### What would you like to be added? / User Story
Extend "Create External CLuster Endpoint" for KubeOne Clusters

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: Extend "Create External CLuster Endpoint" for KubeOne Clusters
**Does this PR close any issues?**:
Fixes #9284

**Special notes for your reviewer**:

**Documentation**:

KubeOne Supported Providers: https://docs.kubermatic.com/kubeone/v1.4/architecture/compatibility/#supported-providers

**Does this PR introduce a user-facing change?**:

```release-note
Add support for KubeOne External Cluster
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/external_cluster.go [hunk 1]
```diff
@@ -73,6 +74,13 @@ type ExternalClusterCloudSpec struct {
 	AKS *ExternalClusterAKSCloudSpec `json:"aks,omitempty"`
 }
 
+// ExternalClusterKubeOneSpec mutually stores access data to a KubeOne cluster.
+type ExternalClusterKubeOneSpec struct {
+	Name                 string                                  `json:"name"`
+	SSHPrivateKey        string                                  `json:"sshPrivateKey"`
```
Comment (zreigz) [inline pkg/apis/kubermatic/v1/external_cluster.go]: use `SSHKey`

File: pkg/handler/v2/external_cluster/external_cluster.go [hunk 2]
```diff
@@ -1008,8 +1017,9 @@ type body struct {
 	// Name is human readable name for the external cluster
 	Name string `json:"name"`
 	// Kubeconfig Base64 encoded kubeconfig
-	Kubeconfig string                          `json:"kubeconfig,omitempty"`
-	Cloud      *apiv2.ExternalClusterCloudSpec `json:"cloud,omitempty"`
+	Kubeconfig    string                          `json:"kubeconfig,omitempty"`
+	SSHPrivateKey string                          `json:"ssh-privatekey,omitempty"`
```
Comment (zreigz) [inline pkg/handler/v2/external_cluster/external_cluster.go]: change to `sshKey`

**Review by zreigz — CHANGES_REQUESTED** [review]

File: pkg/handler/v2/external_cluster/external_cluster.go (hunk 2 above)
Comment (imharshita) [reply pkg/handler/v2/external_cluster/external_cluster.go]: earlier in KKP code, we mentioned if the key is public https://github.com/kubermatic/kubermatic/blob/master/pkg/api/v1/types.go#L468.
WDYT?

Comment (zreigz) [reply pkg/handler/v2/external_cluster/external_cluster.go]: can stay SSHPrivateKey but please change the swagger to `sshPrivateKey`

File: pkg/apis/kubermatic/v1/external_cluster.go [hunk 3]
```diff
@@ -131,5 +131,8 @@ func (i *ExternalCluster) GetCredentialsSecretName() string {
 	if cloud.EKS != nil {
 		cluster.Spec.Cloud.AWS = &AWSCloudSpec{}
 	}
+	if cloud.AKS != nil {
+		cluster.Spec.Cloud.AWS = &AWSCloudSpec{}
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/external_cluster.go]:
```suggestion
		cluster.Spec.Cloud.AKS = &AKSCloudSpec{}
```

copy-paste mistake?

File: pkg/api/v2/types.go [hunk 4]
```diff
@@ -439,9 +439,117 @@ type ExternalClusterSpec struct {
...
+	Passphrase string `json:"passphrase"`
+}
+
+type KubeOneCloudSpec struct {
+	AWS          *KubeOneAWSCloudSpec          `json:"aws,omitempty"`
+	GCP          *KubeOneGCPCloudSpec          `json:"gcp,omitempty"`
+	Azure        *KubeOneAzureCloudSpec        `json:"azure,omitempty"`
+	Digitalocean *KubeOneDigitaloceanCloudSpec `json:"digitalocean,omitempty"`
+	Openstack    *KubeOneOpenstackCloudSpec    `json:"openstack,omitempty"`
+	Packet       *KubeOnePacketCloudSpec       `json:"packet,omitempty"`
```
Comment (kron4eg) [inline pkg/api/v2/types.go]: might be worth to replace Packet to Equinix

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**